src/transformers/models/whisper/modeling_tf_whisper.py

```diff
@@ -120,7 +120,9 @@ def build(self, input_shape):
         super().build(input_shape)
 
     def call(self, input_ids, past_key_values_length=0):
-        return self.weight[past_key_values_length : past_key_values_length + tf.shape(input_ids)[-1]]
+        past_key_values_length = tf.cast(past_key_values_length, tf.int32)
+        gather_indices = tf.range(tf.shape(input_ids)[-1], delta=1) + past_key_values_length
+        return tf.gather(self.weight, gather_indices)
```

Comment:
This strategy makes the shapes known for the XLA compiler

(indexing with `[variable:variable+constant]` yields unknown shapes, even if the variable is the same)